python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py

```diff
@@ -36,17 +41,72 @@ def build_looker_explore_specs(project_dir: Path) -> Sequence[AssetSpec]:
 
 def build_looker_view_specs(project_dir: Path) -> Sequence[AssetSpec]:
     looker_view_specs = []
+    sql_dialect = "bigquery"
 
     # https://cloud.google.com/looker/docs/reference/param-view
     for view_path in project_dir.rglob("*.view.lkml"):
         for view in lkml.load(view_path.read_text()).get("views", []):
-            upstream_table = view.get("sql_table_name") or view["name"]
-            upstream_table_asset_key = AssetKey(upstream_table.replace("`", "").split("."))
+            upstream_tables: Sequence[exp.Table] = [to_table(view["name"], dialect=sql_dialect)]
+
+            # https://cloud.google.com/looker/docs/derived-tables
+            derived_table_sql = view.get("derived_table", {}).get("sql")
+
+            if derived_table_sql and "$" in derived_table_sql:
```

Comment:
Silly example being some column where a user is concatenating a dollar sign string to an integer value to represent a cost.

Reply:
Yeah, this is the tricky part... Looker SQL is potentially parameterized by two differing mechanisms:

1. The `${...}` substitution operator: https://cloud.google.com/looker/docs/sql-and-referring-to-lookml
2. Liquid variable referencing: https://cloud.google.com/looker/docs/liquid-variable-reference

We can probably detect the presence of (1) using regex, instead of a naive `"$" in sql` check.

Unsure how to resolve (2) other than actually rendering the templates with an environment of variables.